### ISSUE TITLE
Fix Request for NonComparableMapKeys: false positives for boxed primitives

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#421]
```
import java.util.HashMap;
import java.util.Map;

public final class Foo {
    final int key = 3;
    void foo() {
        Map<Integer, Integer> map = new HashMap<>();
        map.put(1,2); // <--- false positive
        map.put(key, 4); // <--- false positive
    }
}
```
